### DISCUSSION TITLE
add input box for entering desired information

### DISCUSSION THREAD

**Discussion by asoMansoury** [vpnhood/VpnHood#329] [Ideas]
I think if you add 2 input box as optional for entering the email of user(or telegram ID) and also add another input box for entering any kind of data specially json is really useful, especially for programmers which want to develop on your API's.
can you add this two field?

**Comment by trudyhood** [comment]
Please elaborate more on this. Describe the usage and your business plan that these fields are going to solve.

**Reply by asoMansoury** [reply]
> Please elaborate more on this. Describe the usage and your business plan that these fields are going to solve.

Imagine as a developer I have access to your api which are being called by access server and I want to send email to users by changing server and client's token, instead of telling you to implement this feature I can save my desired data like of email or telegram Id on that field and then customize my Api's based on my requirements.

**Reply by trudyhood** [reply]
Please go into more detail. What kind of information are you going to add to the extra field? Is it such an email address? Why don't they use Token Name as their email address?

* Also, we are working on a plan so you can update your server token seamlessly without sending any emails.
* We will also add a custom field to Access Token so you can put your desired data there.
*  If you plan to use Access Server API, you can store the Access Token ID on your own server and do the extended job there.

**Reply by asoMansoury** [reply]
> Please go into more detail. What kind of information are you going to add to the extra field? Is it such an email address? Why don't they use Token Name as their email address?
> 
>     * Also, we are working on a plan so you can update your server token seamlessly without sending any emails.
> 
>     * We will also add a custom field to Access Token so you can put your desired data there.
> 
>     * If you plan to use Access Server API, you can store the Access Token ID on your own server and do the extended job there.


By Supposing that i can save a JSON file(1MB at maximum) per Token, I will save their phone number on the field, then I will send SMS to clients to signal them any information that is need(It's just one example). This feature will open our hand to customize anything which we think will be useful.

**Reply by trudyhood** [reply]
1MB per token! We will add an extra field, which will be about 200 characters. You need to create your own database and save that information on your own database, or you can find some online storage like Firebase Cloud Storage and store your extra information there.

**Reply by asoMansoury** [reply]
400Character would be enough:)
